### PR TITLE
Adds h-feed support to `tag` lists for #2340

### PR DESCRIPTION
**Pull request by johanbove**
## Here's what I fixed or added:
An h-feed class name is added to the ``<html>`` tag in “content” and “tag” feed views.
The ``<title>`` tag gets the className “p-name”.
The current tag is visible in the ``<title>`` tag's content.

## Here's why I did it:

Proposal to solve #2340 

## Checklist:

- [x] This pull request addresses a single issue
- [ ] If this code includes interface changes, I've included screenshots in this Pull Request thread
- [x] I've adhered to [Known's style guide](http://docs.withknown.com/en/latest/developers/standards/) ([these codesniffer rules](http://docs.withknown.com/en/latest/developers/testing/#code-style-testing) might help!)
- [x] My git branch is named in a descriptive way - i.e., yourname-summary-of-issue
- [x] I've tested my code in-browser
- [x] My code contains descriptive comments
- [ ] I've added tests where applicable, and...
- [ ] I can run the [unit tests](http://docs.withknown.com/en/latest/developers/testing/#unit-testing) successfully.

### CODE REVIEW

File: Idno/Core/Template.php [hunk 1]
```diff
@@ -784,9 +784,8 @@ function getTag()
          * @return bool
          */
         public function isHFeed() {
-            $path = parse_url($_SERVER['REQUEST_URI'], PHP_URL_PATH);
             $classes = \Idno\Core\Idno::site()->template()->getBodyClasses();
```
Comment (johanbove) [inline Idno/Core/Template.php]: Being pragmatic with `template()->getBodyClasses()` here. Hope that is okay.

**Comment by johanbove** [comment]
Received feedback from @jgmac1106 that my changes work.

**Comment by mapkyca** [comment]
Realised I didn't hit "submit review" when I looked at this before...

File: Idno/Core/Template.php [hunk 2]
```diff
@@ -766,6 +766,22 @@ public function getModifiedTS($file)
...
         }
 
+        /**
+         * Checks the current URL for `tag/` and passes this down.
+         * @return string
+         */
+        function getTag()
+        {
+            if ($path = parse_url($_SERVER['REQUEST_URI'], PHP_URL_PATH)) {
+                if ($pos = strpos($path, 'tag/')) {
```
[the remark this answers is not in the document]
Comment (johanbove) [reply Idno/Core/Template.php]: Could change the strpo call to look for ``/tag/`` instead. In the light of the ``h-feed`` logic, the `getTag()` method is a nice-to-have actually.

File: Idno/Core/Template.php [hunk 3]
```diff
@@ -813,6 +830,12 @@ function formatShellVariables($vars)
                 $vars['hidenav'] = \Idno\Core\Idno::site()->embedded();
             }
 
+            if ($tag = \Idno\Core\Idno::site()->template()->getTag()) {
```
[the remark this answers is not in the document]
Comment (johanbove) [reply Idno/Core/Template.php]: It's true that changing the title was not requested in the original issue.  I can change the mutation to put the tag before the existing `title` of course. The `title` tag seemed underused on the "homepage" view as we do change the context of which entries are rendered, but the page `title` did not reflect this. Should I take this out in this PR and move this to a separate PR?

File: templates/default/shell.tpl.php [hunk 4]
```diff
@@ -6,7 +6,7 @@
 
 ?>
 <!DOCTYPE html>
-<html lang="<?php echo $vars['lang']; ?>">
+<html lang="<?php echo $vars['lang']; ?>"<?php if (!empty($template->getTag())) echo ' class="h-feed"' ?>>
```
[the remark this answers is not in the document]
Comment (johanbove) [reply templates/default/shell.tpl.php]: This code is actually outdated. I changed how to check if the `h-feed` tag should be applied or not in later commits. Your comment does still apply as the logic to get to the `title` 's `className` is now done within the `shell.tpl.php`. Probably better to move that logic a level up.

File: templates/default/shell/metatags.tpl.php [hunk 5]
```diff
@@ -1,6 +1,6 @@
 
 <meta charset="utf-8">
-<title><?php echo htmlspecialchars($vars['title']); ?></title>
+<title <?php if (!empty(\Idno\Core\Idno::site()->template()->getTag())) echo 'class="p-name"'?>><?php echo htmlspecialchars($vars['title']); ?></title>
```
[the remark this answers is not in the document]
Comment (johanbove) [reply templates/default/shell/metatags.tpl.php]: Agreeing to using the page vars for this. Will look into making that change.
In regards to having a `class` on a `title`tag, that seems to not bother the HTML5 validators. cf. https://validator.w3.org/nu/?doc=https%3A%2F%2Fsocial.johanbove.info%2F 
The validation check actually is showing worse offenders to "proper" HTML5 code.
The idea to place the `p-name` class attribute on the `title` came is actually suggested here https://indieweb.org/h-feed

File: Idno/Core/Template.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (johanbove) [reply Idno/Core/Template.php]: Changed this to using preg_match() instead so we can do a proper regular expression lookup of the active tag.

File: Idno/Core/Template.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (johanbove) [reply Idno/Core/Template.php]: Changed the code to "prefix" the active tag before the actual title.

File: templates/default/shell.tpl.php (hunk 4 above)
[the remark this answers is not in the document]
Comment (johanbove) [reply templates/default/shell.tpl.php]: Changed this to using page $vars

File: templates/default/shell/metatags.tpl.php (hunk 5 above)
[the remark this answers is not in the document]
Comment (johanbove) [reply templates/default/shell/metatags.tpl.php]: Changed this to using page $vars

**Comment by johanbove** [comment]
@mapkyca Hi Marcus, was wondering if this PR could be still be merged still or I would need to add additional work or abandon this?

**Comment by johanbove** [comment]
Awesome. Thanks!

**Pull request merged**